File: ios/#import <React/BatteryModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BatteryModule, NSObject)

RCT_EXTERN_METHOD(getBatteryLevel: (RCTPromiseResolveBlock)resolve reject: (RCTPromiseRejectBlock)reject)

@end

File: ios/BatteryModule.swift
import Foundation
import React

@objc(BatteryModule)
class BatteryModule: NSObject {
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }
 
    @objc
    func getBatteryLevel(_ resolve: @escaping RCTPromiseResolveBlock,
                         reject: @escaping RCTPromiseRejectBlock) {
        UIDevice.current.isBatteryMonitoringEnabled = true
        let batteryLevel = UIDevice.current.batteryLevel
            print("BatteryModule batteryLevel: \(batteryLevel)")
        if batteryLevel >= 0 {
            resolve(Int(batteryLevel * 100))
        } else {
            let error = NSError(domain: "", code: 200, userInfo: nil)
            reject("E_BATTERY_LEVEL", "Battery level unavailable", error)
        }
    }
}
